FWCore/Catalog/interface/SiteLocalConfig.h

```diff
@@ -30,7 +30,7 @@ namespace edm {
     SiteLocalConfig() {}
     virtual ~SiteLocalConfig() {}
 
-    virtual std::vector<std::string> const& dataCatalogs(void) const = 0;
+    virtual std::vector<std::string> const& dataCatalogs(unsigned catType = 0) const = 0;
```

Comment:
I tried this option without success. The code crashed with complain about "bad allocation". I do not think that 0 and 1 case the returned data is different. They can be both strings. It is just the matter of how trivial catalogs and new (Rucio) catalogs are defined in site_local_config.xml. In trivial catalog (case 0) everything is combined in an url and the codes still need to split it to know the path to storage definition (storage.xml) and the protocol:
"trivialcatalog_file:/cvmfs/cms.cern.ch/SITECONF/T1_US_FNAL/PhEDEx/storage.xml?protocol=xrd"
In the new (Rucio) catalog (case 1), things are more structured in "site", "volume" etc..
<catalog site="T0_CH_CERN" volume="CMSXrootdFederation" protocol="XRootD"/>
Therefore, we can make a string to include all information and the code will split it later as in case 0.
I do not see the benefit of having more complexity of a struct.

Reply:
The struct would be easier to understand and maintain as the meaning different fields would be self-documented than serializing the data into strings. Could you share the code you tried?

I'd actually argue that also in the "case 0" the `dataCatalogs()` returning the full URL (directly from `site_local_config.xml`) is not very good, and it would have been better to do the URL parsing in `SiteLocalConfigService` instead of `FileLocator`. But that is not really worth of improving at this point.

Reply:
Done